packages/core/docs/styleComponents.md

```diff
@@ -0,0 +1,169 @@
+foundation for white label style components) and [Bootstrap Material](https://fezvrasta.github.io/bootstrap-material-design/)
+(a Material Design system based on existing Bootstrap html/class names).
+This is also [a great example of a Design System](https://www.carbondesignsystem.com/components/checkbox/code)
+taking this(style components as lowest abstraction layer) approach.
+  - **well structured**: for clear, readable and maintainable code we use [BEM](http://getbem.com/):
+  it forces the
+  developer to think about component structure in terms of 'block', 'element' and 'modifier'
+  parts/roles. The classes defined in a BEM component form the api for a
+  [subclasser](./definitions.md): he can 'fill in' the BEM selectors by overriding them in the
+  `static get styles()` configuration property on the constructor of a webcomponent.
+  - **flexible markup**: By making the HTML structure purposely 'loose', multiple Design Systems can
+    be made compatible with the style component. When the default html/css structure is not flexible
+    enough, it can be extended by creating new BEM element selectors.
+
+## Why BEM?
+For css class naming, we use [BEM](http://getbem.com/) naming conventions. This helps you identify
+semantics of your markup at a quick glance, resulting in more readable and maintainable code.
+BEM provides us the following advantages:
+- **Clear namings** that convey meaning about semantics and structure at a quick glance
+- **Prevent collisions** within shadow roots. Larger shadow roots using multiple style components
+would easily collide when css selectors would be scoped to the shadow dom in which they were
+created. For instance, calling the header of a card '.header' instead of '.c-card__header'
+would make the card component not reusable in different contexts.
+- **Maximum flexibility** we can write css regardless of context. When we create a style component
+for a card, we can offer it as a style component only and later decide to offer it as a webcomponent.
+Or offer the style component
+as a more lower level api for advanced use cases and the webcomponent (that has limiting markup and
+styling options) for the majority of cases.
+Also, when we already have a webcomponent for the card and people want to create a custom card
+component, it might make more sense to write a new webcomponent instead of extending the existing
+card webcomponent.
+Having style components as described above, allows for maximum flexibility in these scenarios.
+Future CSS updates can then be done from a central place, without having to rewrite forked webcomponents.
+- **Performance** we aim for applications having a limited number of shadow roots and we consider
+shadow root creation a performance concern. Style components reduce the need for shadow roots.
+<!-- TODO: we definitely suffered here in the Polymer era. We might want to create some new perf
+tests for our lit-element based apps to verify this claim again. -->
```

Comment:
maybe make this TODO visible in the Markdown for everyone, not only in the source code?

Reply:
Good idea. Will pick up later